packages/proxy-target/contracts/HifiProxyTarget.sol

```diff
@@ -713,15 +713,25 @@ contract HifiProxyTarget is IHifiProxyTarget {
         (uint256 underlyingReturned, uint256 hTokenReturned) = hifiPool.burn(poolTokensBurned);
 
         // Calculate how much underlying will be redeemed,
-        uint256 underlyingRedeemed = normalize(hTokenReturned, hifiPool.underlyingPrecisionScalar());
+        uint256 underlyingRedeemed = denormalize(hTokenReturned, hifiPool.underlyingPrecisionScalar());
 
         // Redeem the underlying.
         IHToken hToken = hifiPool.hToken();
-        hToken.redeem(hTokenReturned);
+        hToken.redeem(underlyingRedeemed);
```

Comment:
@surbhiaudichya Do we need to check if `underlyingRedeemed` is greater than zero? It is now possible to get `underlyingRedeemed` equal to zero from a dusty amount of `hTokenReturned` due to down-scaling, but it could be a philosophical decision on whether we want this to revert in that edge case or we just go with execution.

Reply:
yes, It was to avoid redundancy.